src/Microsoft.NET.Sdk.Functions.MSBuild/Tasks/Http/DefaultHttpClient.cs

```diff
@@ -1,13 +1,17 @@
 ﻿using System;
 using System.Net.Http;
 using System.Net.Http.Headers;
+using System.Threading;
 using System.Threading.Tasks;
 
 namespace Microsoft.NET.Sdk.Functions.Http
 {
     internal class DefaultHttpClient : IHttpClient, IDisposable
     {
-        private HttpClient _httpClient = new HttpClient();
+        private readonly HttpClient _httpClient = new HttpClient()
+        {
+            Timeout = Timeout.InfiniteTimeSpan
```

Comment:
By switching to the async pattern, do we need to adjust the timeout to infinite?

Reply:
I like the clients to take a call on how long to wait. For example a publish client will wait very differently from a retry client. I still like keeping the default to infinite so that clients have full control on this and can choose to use the default from the server if needed.